### PR TITLE
[TF FE] Enable parallel execution of TensorFlow Layer 2 python tests

### PR DESCRIPTION
**Pull request by Pranshu-S**
Addresses issue: #20919 

### Details:
- Enables parallel execution of TensorFlow Layer 2 python tests
- Fixes test_tf2_keras_conv_lstm_2d.py and test_tf2_map_fn.py to not fail during parallel execution
- Appends args in github workflow to enable parallel execution

Errors fixed:
-  Due to varying Kera activation function addresses causing the workers to get different parameter inputs and thus failing. See [known issue](https://pytest-xdist.readthedocs.io/en/stable/known-limitations.html#order-and-amount-of-test-must-be-consistent)
```
-tensorflow2_keras_tests/test_tf2_keras_conv_lstm_2d.py::TestKerasConvLSTM2D::test_keras_conv_lstm_2d_basic[ ie_device:CPU - precision:FP32 - params:{'params': {'filters': 4, 'kernel_size': (3, 3), 'padding': 'same', 'return_sequences': False, 'activation': <function swish at 0x7f1fadf364d0>}, 'input_shapes': [[2, 5, 20, 30, 2]]} ]
-tensorflow2_keras_tests/test_tf2_keras_conv_lstm_2d.py::TestKerasConvLSTM2D::test_keras_conv_lstm_2d_basic[ ie_device:CPU - precision:FP32 - params:{'params': {'filters': 6, 'kernel_size': (2, 3), 'padding': 'valid', 'dilation_rate': 3, 'recurrent_activation': <function elu at 0x7f1fe6a1a830>, 'return_sequences': True, 'use_bias': True, 'data_format': 'channels_first'}, 'input_shapes': [[2, 5, 1, 40, 30]]} ]
+tensorflow2_keras_tests/test_tf2_keras_conv_lstm_2d.py::TestKerasConvLSTM2D::test_keras_conv_lstm_2d_basic[ ie_device:CPU - precision:FP32 - params:{'params': {'filters': 4, 'kernel_size': (3, 3), 'padding': 'same', 'return_sequences': False, 'activation': <function swish at 0x7f635e4d24d0>}, 'input_shapes': [[2, 5, 20, 30, 2]]} ]
+tensorflow2_keras_tests/test_tf2_keras_conv_lstm_2d.py::TestKerasConvLSTM2D::test_keras_conv_lstm_2d_basic[ ie_device:CPU - precision:FP32 - params:{'params': {'filters': 6, 'kernel_size': (2, 3), 'padding': 'valid', 'dilation_rate': 3, 'recurrent_activation': <function elu at 0x7f6396fa2830>, 'return_sequences': True, 'use_bias': True, 'data_format': 'channels_first'}, 'input_shapes': [[2, 5, 1, 40, 30]]} ]
```

- Due to lambda function definitions giving varying addresses as inputs
```
-tensorflow2_keras_tests/test_tf2_map_fn.py::TestMapFN::test_multiple_inputs_outputs_int32[ ie_device:CPU - precision:FP32 - params:{'fn': <function TestMapFN.<lambda> at 0x7f66c2c63c70>, 'input_type': tf.int32, 'fn_output_signature': (tf.int32, tf.int32, tf.int32), 'back_prop': True, 'input_names': ['x1', 'x2', 'x3'], 'input_shapes': [[2, 1, 3, 4], [2, 1, 3, 4], [2, 1, 3, 4]]} ]
-tensorflow2_keras_tests/test_tf2_map_fn.py::TestMapFN::test_multiple_inputs_outputs_int32[ ie_device:CPU - precision:FP16 - params:{'fn': <function TestMapFN.<lambda> at 0x7f66c2c63c70>, 'input_type': tf.int32, 'fn_output_signature': (tf.int32, tf.int32, tf.int32), 'back_prop': True, 'input_names': ['x1', 'x2', 'x3'], 'input_shapes': [[2, 1, 3, 4], [2, 1, 3, 4], [2, 1, 3, 4]]} ]
+tensorflow2_keras_tests/test_tf2_map_fn.py::TestMapFN::test_multiple_inputs_outputs_int32[ ie_device:CPU - precision:FP32 - params:{'fn': <function TestMapFN.<lambda> at 0x7f211b56fd00>, 'input_type': tf.int32, 'fn_output_signature': (tf.int32, tf.int32, tf.int32), 'back_prop': True, 'input_names': ['x1', 'x2', 'x3'], 'input_shapes': [[2, 1, 3, 4], [2, 1, 3, 4], [2, 1, 3, 4]]} ]
+tensorflow2_keras_tests/test_tf2_map_fn.py::TestMapFN::test_multiple_inputs_outputs_int32[ ie_device:CPU - precision:FP16 - params:{'fn': <function TestMapFN.<lambda> at 0x7f211b56fd00>, 'input_type': tf.int32, 'fn_output_signature': (tf.int32, tf.int32, tf.int32), 'back_prop': True, 'input_names': ['x1', 'x2', 'x3'], 'input_shapes': [[2, 1, 3, 4], [2, 1, 3, 4], [2, 1, 3, 4]]} ]
```

### CODE REVIEW

File: tests/layer_tests/tensorflow2_keras_tests/test_tf2_map_fn.py [hunk 1]
```diff
@@ -97,17 +115,17 @@ def test_multiple_outputs(self, params, ie_device, precision, ir_version, temp_d
...
              fn_output_signature=tf.int32, back_prop=True,
              input_names=["x1", "x2", "x3"],
              input_shapes=[[2, 1, 3], [2, 1, 3], [2, 1, 3]]),
-        pytest.param(dict(fn=lambda x: (x[0] + x[1] + x[2], x[0] - x[2] + x[1], 2 + x[2]),
+        pytest.param(dict(fn=fn_6,
                           input_type=tf.int32,
                           fn_output_signature=(tf.int32, tf.int32, tf.int32), back_prop=True,
                           input_names=["x1", "x2", "x3"],
                           input_shapes=[[2, 1, 3, 4], [2, 1, 3, 4], [2, 1, 3, 4]]),
-                     marks=[pytest.mark.xfail(reason="61587"), pytest.mark.precommit_tf_fe])
```
Comment (rkazants) [inline tests/layer_tests/tensorflow2_keras_tests/test_tf2_map_fn.py:110]: please return this case into `precommit`

**Comment by rkazants** [comment]
build_jenkins

**Comment by Pranshu-S** [comment]
Hey @rkazants 
Raised a update for this. Do take a look

**Comment by Pranshu-S** [comment]
Hi, 
Quick qs: What does adding of `pytest.mark.precommit_tf_fe]`  mark does in a unit test? [Ref commit](https://github.com/openvinotoolkit/openvino/pull/23344/commits/0ae6e246756eb25bbeeac03e693628399af0437d#diff-182869b4dd11069d9987f3d7ce58c202add32ef6c73ecb858e6fd1ca361dfbf6R128) 

This seemed to have worked in my dev setup without this commit. However, after passing it into the precommit - it has started to fail. After studying the failed build - it seems to be related to this particular test case wherein the function addresses seem to differ in the different worker threads

For reference from logs - 
```
-tensorflow2_keras_tests/test_tf2_map_fn.py::TestMapFN::test_multiple_inputs_outputs_int32[ ie_device:CPU - precision:FP16 - params:{'fn': <function fn_6 at 0xffff2e527060>, 'input_type': tf.int32, 'fn_output_signature': (tf.int32, tf.int32, tf.int32), 'back_prop': True, 'input_names': ['x1', 'x2', 'x3'], 'input_shapes': [[2, 1, 3, 4], [2, 1, 3, 4], [2, 1, 3, 4]]} ]
+tensorflow2_keras_tests/test_tf2_map_fn.py::TestMapFN::test_multiple_inputs_outputs_int32[ ie_device:CPU - precision:FP16 - params:{'fn': <function fn_6 at 0xffff05de2480>, 'input_type': tf.int32, 'fn_output_signature': (tf.int32, tf.int32, tf.int32), 'back_prop': True, 'input_names': ['x1', 'x2', 'x3'], 'input_shapes': [[2, 1, 3, 4], [2, 1, 3, 4], [2, 1, 3, 4]]} ]
```

Wanted to get a high level understanding before taking a deeper look. Thanks!

**Comment by rkazants** [comment]
> hat does adding of `pytest.mark.precommit_tf_fe]` mark does in a unit test? [Ref commit](https://github.com/openvinotoolkit/openvino/pull/23344/commits/0ae6e246756eb25bbeeac03e693628399af0437d#diff-182869b4dd11069d9987f3d7ce58c202add32ef6c73ecb858e6fd1ca361dfbf6R128)
> 
> This seemed to have worked in my dev setup without this commit. However, after passing it into the precommit - it has started to fail. After studying the failed build - it seems to be related to this particular test case wherein the function addresses seem to differ in the different worker threads
> 
> For reference from logs -

`precommit_tf_fe` is marker of precommit test cases to test new implementation for TensorFlow model support, just to differentiate test cases for legacy implementation of TF support and new logic.

**Comment by rkazants** [comment]
@Pranshu-S, please run TF2 layer tests locally on your machine in parallel mode. 
Now tests look failing.

**Comment by Pranshu-S** [comment]
Hi,
Can we try to build the changes? It should be good to go.

Thanks!

**Comment by mlukasze** [comment]
build_jenkins

**Comment by Pranshu-S** [comment]
Hey @mlukasze @rkazants 
I think the tests are not failing anymore. Anything I need to do further on this PR?

**Comment by rkazants** [comment]
build_jenkins

**Review by rkazants — APPROVED** [review]

**Comment by rkazants** [comment]
Acceleration on Linux is ~2.5 times.

**Pull request merged**